### PR TITLE
feat: improve gas consumption for CBridgeFacetPacked

### PR DESCRIPTION
**Pull request by ezynda3**

### CODE REVIEW

File: src/Facets/CBridgeFacetPacked.sol [hunk 1]
```diff
@@ -101,83 +216,53 @@ contract CBridgeFacetPacked is ILiFi {
...
+        bytes calldata _data
+    )
+        external
+        pure
+        returns (BridgeData memory, CBridgeFacet.CBridgeData memory)
+    {
         require(
-            msg.data.length >= 96,
-            "callData length smaller than required"
+            _data.length >= 40,
```
Comment (maxklenk) [inline src/Facets/CBridgeFacetPacked.sol:236]: Why 40? we are reading up to 60 below.

File: src/Facets/CBridgeFacetPacked.sol [hunk 2]
```diff
@@ -200,93 +285,35 @@ contract CBridgeFacetPacked is ILiFi {
...
-        );
-    }
+    function decode_startBridgeTokensViaCBridgeERC20Packed(
+        bytes calldata _data
+    )
+        external
+        pure
+        returns (BridgeData memory, CBridgeFacet.CBridgeData memory)
+    {
+        require(_data.length >= 76, "data passed is not the correct length");
```
Comment (maxklenk) [inline src/Facets/CBridgeFacetPacked.sol:304]: 76 looks too long

**Review by maxklenk — CHANGES_REQUESTED** [review]
Great. Tests for the decoders would be good to add.

File: test/solidity/Facets/HopFacetPackedL1.t.sol [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+            amountUSDC,
+            amountOutMinUSDC,
+            address(0),
+            0,
+            HOP_USDC_BRIDGE
+        );
+        vm.stopPrank();
+    }
+
+    // Encode
```
Comment (0xDEnYO) [inline test/solidity/Facets/HopFacetPackedL1.t.sol:327]: Why is this all commented out?
Can it be removed?

File: test/solidity/Facets/HopFacetPackedL2.t.sol [hunk 4]
```diff
@@ -0,0 +1,329 @@
+pragma solidity 0.8.17;
+
```
Comment (0xDEnYO) [inline test/solidity/Facets/HopFacetPackedL2.t.sol:2]: All of my comments on test/solidity/Facets/HopFacetPackedL1.t.sol seem to apply here, too

File: test/solidity/Gas/CBridgeFacetPackedARB.gas.t.sol [hunk 5]
```diff
@@ -49,7 +49,8 @@ contract CBridgeGasTest is Test, DiamondTest {
         /// Perpare CBridgeFacetPacked
```
Comment (0xDEnYO) [inline test/solidity/Gas/CBridgeFacetPackedARB.gas.t.sol:49]: typo

File: test/solidity/Gas/CBridgeFacetPackedARB.gas.t.sol [hunk 6]
```diff
@@ -206,12 +159,21 @@ contract CBridgeGasTest is Test, DiamondTest {
         vm.stopPrank();
     }
 
+    function testStartBridgeTokensViaCBridgeERC20Packed_StandAlone() public {
```
Comment (0xDEnYO) [inline test/solidity/Gas/CBridgeFacetPackedARB.gas.t.sol:162]: My comment about checking events in test cases also applies here

**Pull request merged**